sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -423,7 +423,14 @@ case class InsertIntoHiveTable(
     // Attempt to delete the staging directory and the inclusive files. If failed, the files are
     // expected to be dropped at the normal termination of VM since deleteOnExit is used.
     try {
-      createdTempDir.foreach { path => path.getFileSystem(hadoopConf).delete(path, true) }
+      createdTempDir.foreach { path =>
+        val fs = path.getFileSystem(hadoopConf)
+        fs.delete(path, true)
```

Comment:
`FileSystem.delete returns true if delete is successful else false.` So it makes sense to re-write a bit
```
if (fs.delete(path, true)) {
fs.cancelDeleteOnExit(path)
}
```